src/main/java/com/amazon/ion/impl/lite/IonSystemLite.java

```diff
@@ -767,17 +762,7 @@ private boolean cloneAnnotationsIfAny(IonValueLite value) {
             if (annotations.length == 0) {
                 return false;
             }
-
-            for (int i = 0; i < annotations.length; i++)
-            {
-                SymbolToken token = annotations[i];
-                String text = token.getText();
-                if (text != null && token.getSid() != UNKNOWN_SYMBOL_ID )
-                {
-                    annotations[i] = newSymbolToken(text, UNKNOWN_SYMBOL_ID);
-                }
-            }
-            value.setTypeAnnotationSymbols(annotations);
+            value.setTypeAnnotationSymbolsNoCopy(annotations);
```

Comment:
This is safe only if `IonReader.getTypeAnnotationSymbols()` already returns a *copy* of the reader's internal annotations array. It doesn't look like this is a documented contract. Is it true for all implementations in all cases?

Note: even if we can't go as far as avoiding an array copy, we can still make the change that doesn't copy re-allocate each individual symbol token to remove the symbol IDs.